x-pack/plugins/triggers_actions_ui/public/application/components/builtin_action_types/xmatters/xmatters_connectors.tsx

```diff
@@ -0,0 +1,250 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useEffect, useState } from 'react';
+import { FormattedMessage } from '@kbn/i18n-react';
+
+import {
+  EuiFieldPassword,
+  EuiFieldText,
+  EuiFormRow,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiSpacer,
+  EuiTitle,
+  EuiButtonGroup,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { ActionConnectorFieldsProps } from '../../../../types';
+import { XmattersActionConnector, XmattersAuthenticationType } from '../types';
+
+const XmattersActionConnectorFields: React.FunctionComponent<
+  ActionConnectorFieldsProps<XmattersActionConnector>
+> = ({ action, editActionConfig, editActionSecrets, errors, readOnly }) => {
+  const { user, password, urlSecrets } = action.secrets;
+  const { urlConfig, usesBasic } = action.config;
+
+  useEffect(() => {
+    if (!action.id) {
+      editActionConfig('usesBasic', true);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const isUrlInvalid: boolean =
+    errors.urlSecrets !== undefined &&
+    errors.urlSecrets.length > 0 &&
+    urlSecrets !== undefined &&
+    errors.urlConfig !== undefined &&
+    errors.urlConfig.length > 0 &&
+    urlConfig !== undefined;
+  const isPasswordInvalid: boolean =
+    password !== undefined && errors.password !== undefined && errors.password.length > 0;
+  const isUserInvalid: boolean =
+    user !== undefined && errors.user !== undefined && errors.user.length > 0;
+
+  const authenticationButtons = [
+    { id: XmattersAuthenticationType.Basic, label: 'Basic Authentication' },
+    { id: XmattersAuthenticationType.URL, label: 'URL Authentication' },
+  ];
+
+  const [selectedAuth, setSelectedAuth] = useState(XmattersAuthenticationType.Basic);
+
+  return (
+    <>
+      <EuiTitle size="xxs">
+        <h4>
+          <FormattedMessage
+            id="xpack.triggersActionsUI.components.builtinActionTypes.xmattersAction.authenticationLabel"
+            defaultMessage="Authentication"
+          />
+        </h4>
+      </EuiTitle>
+      <EuiSpacer size="xs" />
+      <EuiFormRow fullWidth>
+        <p>
+          <FormattedMessage
+            id="xpack.triggersActionsUI.components.builtinActionTypes.xmattersAction.connectorSettingsLabel"
+            defaultMessage="Select the authentication method used when setting up the xMatters trigger."
+          />
+        </p>
+      </EuiFormRow>
+      <EuiSpacer size="l" />
+      <EuiButtonGroup
+        isFullWidth
+        buttonSize="m"
+        legend="Basic Authentication"
+        options={authenticationButtons}
+        color="primary"
+        idSelected={selectedAuth}
+        onChange={(id: string) => {
+          if (id === 'Basic Authentication') {
```

Comment:
I think the check should be `id === XmattersAuthenticationType.Basic` because the `EuiButtonGroup` returns the ids and not the labels.